Neos.Cache/Classes/Backend/SetupEnabledBackendInterface.php

```diff
@@ -0,0 +1,38 @@
+<?php
+namespace Neos\Cache\Backend;
+
+/*
+ * This file is part of the Neos.Cache package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Error\Messages\Result;
+
+/**
+ * A contract for a Cache Backend which allows to be setup
+ *
+ * @api
+ */
+interface SetupEnabledBackendInterface extends BackendInterface
+{
+    /**
+     * TODO document
+     *
+     * @return Result
+     * @api
+     */
+    public function setup(): Result;
+
+    /**
+     * TODO document
+     *
+     * @return Result
+     * @api
+     */
+    public function getStatus(): Result;
```

Comment:
getStatus outside of the interface (so in the implementation) might be a bit unspecific. Could also be the current status of the cache (or cache content) or any other kind of status. So in that case I would probably name it getSetupStatus ? Even if that sounds a bit awkward.
But I could very well imagine having a getStatus to ask if the cache is available right now (eg. for redis) at a later point.

Reply:
I agree but on the other hand:
I played around with the RedisBackend and implemented the `getStatus()` to render details about the server version etc. It's quite useful IMO.
So **suggestion:**
I split the interface into `BackendWithSetupInterface` and `BackendWithStatusInterface`. Some like the PdoBackend can implement both and the ones that can't be set up via PHP (i.e. Redis) can only implement the status interface (and we can at any point extend `getStatus()` to return more infos about the cache).
Do you agree to that approach?